corehq/apps/userreports/util.py

```diff
@@ -355,3 +356,26 @@ def unescape(value):
         # double-unescape because corehq.apps.userreports.util.get_table_name escapes twice
         return unescape(unescape(match.group(1)))
     raise ValueError(f"Expected {table_name} to start with {UCR_TABLE_PREFIX} or {LEGACY_UCR_TABLE_PREFIX}")
+
+
+def register_data_source_change(domain, data_source_id, row_changes, action):
+    from corehq.motech.repeaters.models import DataSourceRepeater
+    from corehq.motech.repeaters.signals import ucr_data_source_updated
+    try:
+        if not (
```

Comment:
Can we move this if block to where the function is getting called since this is at the top of the function definition?

Reply:
The only reason why I put it in here is to reduce code duplication, since this function is called from two places. If you feel strongly about this, I'm happy to move it..

Reply:
That makes sense.
I like it when the check for feature flag is exposed as early as possible.
So, if it makes sense, at least the feature flag check can be moved to places where the method is being called to just highlight it to someone, if they are reading code, earlier, and not when they visit this method.